File: Eunleelee/2024-02/2월 28일/프로세스.py
def solution(priorities, location):
    answer = 0; n = max(priorities)
    i = 0
    an = 0
    while 1:
        if priorities[i] < n:
            priorities.append(priorities[i])
            priorities[i] = 0
            if i == location:
                location = len(priorities)-1
        elif priorities[i] >= n:
            priorities[i] = 0
            answer += 1
            n = max(priorities)
            if i == location:
                an = -1
        if an == -1:
            break
        i += 1

            
    return answer

File: wooseongshin/2월/28일/프로세스.java
import java.util.*;
class Solution {

    public int solution(int[] priorities, int location) {

        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());

        for(int i=0; i< priorities.length; i++){
            pq.add(priorities[i]);
        }
        int answer = 0;

        while(!pq.isEmpty()) {
            for (int i=0; i<priorities.length; i++) {
                if (priorities[i] == pq.peek()){
                    if(i == location){
                        answer++;
                        pq.poll();
                        return answer;
                    }
                    pq.poll();
                    answer++;
                }
            }
        }

        return -1;
    }
}
